asset-transfer-private-data/chaincode-go/privateAssetTransfer.go

```diff
@@ -0,0 +1,469 @@
+	}
+
+	// Make submitting client the owner
+	asset := &Asset{
+		Type: assetInput.Type,
+		ID:       	assetInput.ID,
+		Color:      assetInput.Color,
+		Size:       assetInput.Size,
+		Owner:      clientID,
+	}
+	assetJSONasBytes, err := json.Marshal(asset)
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+
+	// Save asset to private data collection
+	err = ctx.GetStub().PutPrivateData(assetCollection, assetInput.ID, assetJSONasBytes)
+	if err != nil {
+		return fmt.Errorf("failed to put Asset: %s", err.Error())
+	}
+
+	// Save asset details to collection visible to owning organization
+	assetPrivateDetails := &AssetPrivateDetails{
+		ID:       			 	assetInput.ID,
+		AppraisedValue:   assetInput.AppraisedValue,
+	}
+
+	assetPrivateDetailsAsBytes, err := json.Marshal(assetPrivateDetails) // marshal asset details to JSON
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+
+	// get collection name for this organization
+	orgCollection, err := getCollectionName(ctx)
+
+	// put asset into owners org specific private data collection
+	err = ctx.GetStub().PutPrivateData(orgCollection, assetInput.ID, assetPrivateDetailsAsBytes)
+	if err != nil {
+		return fmt.Errorf("failed to put asset private details: %s", err.Error())
+	}
+	return nil
+}
+
+// AgreeToTransfer is used by the potential buyer of the asset to agree to the
+// asset price. The agreed to appraisal value is stored in the buying orgs
+// org specifc collection, while the the buyer client ID is stored in the asset collection
+// using a composite key
+func (s *SmartContract) AgreeToTransfer(ctx contractapi.TransactionContextInterface, assetID string) error {
+
+	// Get ID of submitting client identity
+	clientID, err := ctx.GetClientIdentity().GetID()
+	if err != nil {
+		return fmt.Errorf("failed to get verified OrgID: %s", err.Error())
+	}
+
+	// price is private, therefore it gets passed in transient field
+	transMap, err := ctx.GetStub().GetTransient()
+	if err != nil {
+		return fmt.Errorf("Error getting transient: " + err.Error())
+	}
+
+	// persist the JSON bytes as-is so that there is no risk of nondeterministic marshaling.
+	priceJSON, ok := transMap["asset_value"]
+	if !ok {
+		return fmt.Errorf("asset_value key not found in the transient map")
+	}
+
+	// get collection name for this organization
+	orgCollection, err := getCollectionName(ctx)
+
+	// put price in the org specifc private data collection
+	err = ctx.GetStub().PutPrivateData(orgCollection, assetID, priceJSON)
+	if err != nil {
+		return fmt.Errorf("failed to put asset bid: %s", err.Error())
+	}
+
+	// create agreeement where you indicate the identity has agreed to purchase
+	transferAgreeKey, err := ctx.GetStub().CreateCompositeKey("transferAgreement",[]string{assetID})
+	if err != nil {
+		return fmt.Errorf("failed to create composite key: %s", err.Error())
+	}
+
+	err = ctx.GetStub().PutPrivateData(assetCollection, transferAgreeKey, []byte(clientID))
```

Comment:
I agree that it is not realistic, but we discussed having a transfer agreement recorded so that the ID did not have to be passed out of band (and the ReturnID) function could be removed. 

I think with the secured asset example already in place, we can keep this example less realistic. However, still open for suggests simple flow would provide a good example.